quantum/via.h

```diff
@@ -58,7 +58,35 @@
 
 // This is changed only when the command IDs change,
 // so VIA Configurator can detect compatible firmware.
-#define VIA_PROTOCOL_VERSION 0x000A
+#define VIA_PROTOCOL_VERSION 0x000B
+
+// This is a version number for the firmware for the keyboard.
+// It can be used to ensure the VIA keyboard definition and the firmware
+// have the same version, especially if there are changes to custom values.
+// Define this in config.h to override and bump this number.
+// This is *not* required if the keyboard is only using basic functionality
+// and not using custom values for lighting, rotary encoders, etc.
+#ifndef VIA_FIRMWARE_VERSION
+#    define VIA_FIRMWARE_VERSION 0x00000000
+#endif
+
+// Enable the VIA command handlers for QMK Core features if those features
+// are enabled.
+#if defined(BACKLIGHT_ENABLE)
+#    define VIA_QMK_BACKLIGHT_ENABLE
+#endif
+
+#if defined(RGBLIGHT_ENABLE)
+#    define VIA_QMK_RGBLIGHT_ENABLE
+#endif
+
+#if defined(RGB_MATRIX_ENABLE)
+#    define VIA_QMK_RGB_MATRIX_ENABLE
+#endif
+
+#if defined(AUDIO_ENABLE)
+#    define VIA_QMK_AUDIO_ENABLE
+#endif
```

Comment:
At this point, aren't these redundant?  It would be simpler to just check for the normal defines rather than the the via specific ones. 

```suggestion
```